### PR TITLE
Add regression test and fix for overflow waiting error

### LINKED ISSUE #173: OverflowError: int too large to convert to float
## Background
The `SessionController` class `proceed_to_and_wait_for` retries attempts to proceed to a particular stage while the status of the attempt is not completed or errored. This can be the case when asynchronous processing of a sync stage has been implemented, during [legacy handling](https://github.com/learningequality/morango/blob/release-v0.6.x/morango/api/viewsets.py#L427) of 0.14 instances syncing, and during [transferring](https://github.com/learningequality/morango/blob/release-v0.6.x/morango/sync/operations.py#L1621) of a large amount of data.

## Details
```
Traceback (most recent call last):
  File "/home/casey/.pyenv/versions/15Jun2022-vf/bin/kolibri", line 11, in <module>
    sys.exit(main())
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/utils/cli.py", line 200, in invoke
    return super(KolibriDjangoCommand, self).invoke(ctx)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/utils/cli.py", line 368, in manage
    execute_from_command_line(["kolibri manage"] + ctx.args)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/core/tasks/management/commands/base.py", line 119, in handle
    return self.handle_async(*args, **options)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/core/auth/management/commands/resumesync.py", line 57, in handle_async
    self._sync(sync_session_client, **options)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/core/auth/management/utils.py", line 442, in _sync
    push_filter,
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/core/auth/management/utils.py", line 584, in _push
    sync_client.run()
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/morango/sync/syncsession.py", line 653, in run
    transfer_stages.TRANSFERRING, callback=status.in_progress.fire
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/morango/sync/syncsession.py", line 613, in proceed_to_and_wait_for
    result = self.controller.proceed_to_and_wait_for(stage, callback=callback)
  File "/home/casey/.pyenv/versions/3.6.15/envs/15Jun2022-vf/lib/python3.6/site-packages/kolibri/dist/morango/sync/controller.py", line 214, in proceed_to_and_wait_for
    sleep(min(0.3 * (2 ** tries - 1), max_interval))
OverflowError: int too large to convert to float
```

### PR DESCRIPTION
**Pull request by bjester**
## Summary
- @AllanOXDi and I co-hacked on this fix
- We solved for that maximum tries that would cause the sleep time to always be `max_interval` to avoid causing overflow errors when multiplying a float with a large number

## TODO

- [X] Have tests been written for the new code?
- [ ] Has documentation been written/updated?
- [ ] New dependencies (if any) added to requirements file

## Reviewer guidance
- Run the tests-- do they pass?
- Check my math-- did I determine `max_interval_tries` correctly?

## Issues addressed

Fixes https://github.com/learningequality/morango/issues/173

### CODE REVIEW

File: morango/sync/controller.py [hunk 1]
```diff
@@ -207,11 +208,16 @@ def proceed_to_and_wait_for(
         tries = 0
         context = context or self.context
         max_interval = max_interval or context.max_backoff_interval
+        # solve for the number of tries at which our sleep time will always be max_interval
+        max_interval_tries = round(1 + ((math.log(max_interval) - math.log(0.3)) / math.log(2)))
```
Comment (bjester) [inline morango/sync/controller.py]: I think `math.ceil` is what I want here, not `round`

Comment (rtibbles) [reply morango/sync/controller.py]: I think you might be able to leave out the `1+` as well if you use `math.ceil`? From some quick calculation checks, it looked like the value was correct without the added `1` if it was rounded up.

File: morango/sync/controller.py [hunk 2]
```diff
@@ -207,11 +208,16 @@ def proceed_to_and_wait_for(
...
 
         while result not in transfer_statuses.FINISHED_STATES:
             if tries > 0:
                 # exponential backoff up to max_interval
-                sleep(min(0.3 * (2 ** tries - 1), max_interval))
+                if tries >= max_interval_tries:
+                    sleep(max_interval)
+                else:
+                    sleep(0.3 * (2 ** tries - 1))
```
Comment (rtibbles) [inline morango/sync/controller.py:220]: Confirmed with some local empirical testing that these values should continue to be coerceable to `int` even if `max_interval` is so enormously large, the sleep would take forever.

**Review by rtibbles — COMMENTED** [review]
Tests make sense, and seem to provide a solid regression test. Some quick empirical checks of the Maths seems like it is working as intended, but it seemed like maybe the `1+` could be discarded in favour of `ceil`, as you suggested.

**Comment by bjester** [comment]
@rtibbles I see now I misinterpreted the `tries - 1` as being the exponent, not just `tries`, and that's where the `1 + ` came from. Fixing that, and changing to `ceil` should be correct
